### ISSUE TITLE
Errors in dispatch_web_1 container

### ISSUE THREAD

**Issue by sfc-gh-pkommini** [Netflix/dispatch-docker#87]
I did the following as recommended:

```bash
./install.sh
docker-compose up -d
# then went to UI and registered my email ID and password and logged in
```
I got these logs off of the `dispatch_web_1` container
**OS**: MacOS High Sierra: 10.15.7
**Docker Version**: 19.03.13

**Error 1:**
```
[SQL: SELECT workflow_instance.resource_type AS workflow_instance_resource_type, workflow_instance.resource_id AS workflow_instance_resource_id, workflow_instance.weblink AS workflow_instance_weblink, workflow_instance.id AS workflow_instance_id, workflow_instance.workflow_id AS workflow_instance_workflow_id, workflow_instance.incident_id AS workflow_instance_incident_id, workflow_instance.parameters AS workflow_instance_parameters, workflow_instance.run_reason AS workflow_instance_run_reason, workflow_instance.creator_id AS workflow_instance_creator_id, workflow_instance.status AS workflow_instance_status, workflow_instance.created_at AS workflow_instance_created_at, workflow_instance.updated_at AS workflow_instance_updated_at 
FROM workflow_instance 
WHERE %(param_1)s = workflow_instance.incident_id]
[parameters: {'param_1': 1}]
(Background on this error at: http://sqlalche.me/e/13/f405)
ERROR:uvicorn.error:Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/http/h11_impl.py", line 389, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 26, in __call__
    await response(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 228, in __call__
    await run_until_first_complete(
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in run_until_first_complete
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in <listcomp>
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 220, in stream_response
    async for chunk in self.body_iterator:
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 56, in body_stream
    task.result()
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 38, in coro
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/sentry_asgi/middleware.py", line 22, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/sentry_asgi/middleware.py", line 19, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 26, in __call__
    await response(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 228, in __call__
    await run_until_first_complete(
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in run_until_first_complete
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in <listcomp>
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 220, in stream_response
    async for chunk in self.body_iterator:
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 56, in body_stream
    task.result()
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 38, in coro
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 26, in __call__
    await response(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 228, in __call__
    await run_until_first_complete(
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in run_until_first_complete
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in <listcomp>
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 220, in stream_response
    async for chunk in self.body_iterator:
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 56, in body_stream
    task.result()
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 38, in coro
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 376, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 179, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 190, in app
    response_data = await serialize_response(
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 103, in serialize_response
    value, errors_ = await run_in_threadpool(
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 34, in run_in_threadpool
    return await loop.run_in_executor(None, func, *args)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "pydantic/fields.py", line 579, in pydantic.fields.ModelField.validate
  File "pydantic/fields.py", line 738, in pydantic.fields.ModelField._validate_singleton
  File "pydantic/fields.py", line 745, in pydantic.fields.ModelField._apply_validators
  File "pydantic/class_validators.py", line 310, in pydantic.class_validators._generic_validator_basic.lambda12
  File "pydantic/main.py", line 658, in pydantic.main.BaseModel.validate
  File "pydantic/main.py", line 360, in pydantic.main.BaseModel.__init__
  File "pydantic/main.py", line 981, in pydantic.main.validate_model
  File "pydantic/fields.py", line 590, in pydantic.fields.ModelField.validate
  File "pydantic/fields.py", line 621, in pydantic.fields.ModelField._validate_sequence_like
  File "pydantic/fields.py", line 731, in pydantic.fields.ModelField._validate_singleton
  File "pydantic/fields.py", line 579, in pydantic.fields.ModelField.validate
  File "pydantic/fields.py", line 738, in pydantic.fields.ModelField._validate_singleton
  File "pydantic/fields.py", line 745, in pydantic.fields.ModelField._apply_validators
  File "pydantic/class_validators.py", line 310, in pydantic.class_validators._generic_validator_basic.lambda12
  File "pydantic/main.py", line 662, in pydantic.main.BaseModel.validate
  File "pydantic/main.py", line 566, in pydantic.main.BaseModel.from_orm
  File "pydantic/main.py", line 960, in pydantic.main.validate_model
  File "pydantic/utils.py", line 412, in pydantic.utils.GetterDict.get
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 287, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/attributes.py", line 723, in get
    value = self.callable_(state, passive)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/strategies.py", line 759, in _load_for_state
    return self._emit_lazyload(
  File "<string>", line 1, in <lambda>
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/strategies.py", line 900, in _emit_lazyload
    q(session)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/ext/baked.py", line 544, in all
    return list(self)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/ext/baked.py", line 444, in __iter__
    return q._execute_and_instances(context)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3560, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1124, in _execute_clauseelement
    ret = self._execute_context(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception
    util.raise_(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "workflow_instance" does not exist
LINE 2: FROM workflow_instance 
```
**Error 2:**
```
[SQL: SELECT workflow_instance.resource_type AS workflow_instance_resource_type, workflow_instance.resource_id AS workflow_instance_resource_id, workflow_instance.weblink AS workflow_instance_weblink, workflow_instance.id AS workflow_instance_id, workflow_instance.workflow_id AS workflow_instance_workflow_id, workflow_instance.incident_id AS workflow_instance_incident_id, workflow_instance.parameters AS workflow_instance_parameters, workflow_instance.run_reason AS workflow_instance_run_reason, workflow_instance.creator_id AS workflow_instance_creator_id, workflow_instance.status AS workflow_instance_status, workflow_instance.created_at AS workflow_instance_created_at, workflow_instance.updated_at AS workflow_instance_updated_at 
FROM workflow_instance 
WHERE %(param_1)s = workflow_instance.incident_id]
[parameters: {'param_1': 1}]
(Background on this error at: http://sqlalche.me/e/13/f405)
INFO:     192.168.32.1:52150 - "GET /api/v1/tags/ HTTP/1.1" 500 Internal Server Error
INFO:     192.168.32.1:52156 - "GET /api/v1/incident_priorities/?sortBy[]=view_order&descending[]=false HTTP/1.1" 200 OK
INFO:     192.168.32.1:52152 - "GET /api/v1/documents/?field[]=resource_type&op[]=%3D%3D&value[]=dispatch-incident-faq HTTP/1.1" 200 OK
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/http/h11_impl.py", line 389, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 26, in __call__
    await response(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 228, in __call__
    await run_until_first_complete(
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in run_until_first_complete
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in <listcomp>
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 220, in stream_response
    async for chunk in self.body_iterator:
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 56, in body_stream
    task.result()
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 38, in coro
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/sentry_asgi/middleware.py", line 22, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/sentry_asgi/middleware.py", line 19, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 26, in __call__
    await response(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 228, in __call__
    await run_until_first_complete(
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in run_until_first_complete
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in <listcomp>
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 220, in stream_response
    async for chunk in self.body_iterator:
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 56, in body_stream
    task.result()
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 38, in coro
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 26, in __call__
    await response(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 228, in __call__
    await run_until_first_complete(
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in run_until_first_complete
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 18, in <listcomp>
    [task.result() for task in done]
  File "/usr/local/lib/python3.8/site-packages/starlette/responses.py", line 220, in stream_response
    async for chunk in self.body_iterator:
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 56, in body_stream
    task.result()
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/base.py", line 38, in coro
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 376, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 179, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 111, in __call__
    await self.middleware_stack(scope, receive, send)
INFO:     192.168.32.1:52154 - "GET /api/v1/plugins/?itemsPerPage=-1&field[]=enabled&op[]=%3D%3D&value[]=true HTTP/1.1" 200 OK
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 566, in __call__
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 227, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 41, in app
    response = await func(request)
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 182, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 135, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 34, in run_in_threadpool
    return await loop.run_in_executor(None, func, *args)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.8/site-packages/dispatch/tag/views.py", line 34, in get_tags
    return search_filter_sort_paginate(
  File "/usr/local/lib/python3.8/site-packages/dispatch/database.py", line 211, in search_filter_sort_paginate
    query, pagination = apply_pagination(query, page_number=page, page_size=items_per_page)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy_filters/pagination.py", line 42, in apply_pagination
    total_results = query.count()
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3803, in count
    return self.from_self(col).scalar()
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3523, in scalar
    ret = self.one()
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3490, in one
    ret = self.one_or_none()
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3459, in one_or_none
    ret = list(self)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3535, in __iter__
    return self._execute_and_instances(context)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/query.py", line 3560, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1124, in _execute_clauseelement
    ret = self._execute_context(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception
    util.raise_(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column tag.tag_type_id does not exist
LINE 2: ...tag_source, tag.discoverable AS tag_discoverable, tag.tag_ty...
```
I was able to install and then register new user, but after that I get all sorts of errors just loading the home page. Web UI says something went wrong and I should contact the administrator.
![image](https://user-images.githubusercontent.com/72515998/99484764-f49e5000-2915-11eb-872c-fbf2159b9683.png)

I also tried creating an incident and got errors saying that fields were missing although they weren't marked as required fields in the UI. But then I didn't setup all the plugins, so I guess this is because of that, although it would be great if it failed gracefully.

**Comment by kevgliss** [comment]
Is this a new database or on an existing database? The errors you point out are due to new database migrations in the last few versions. 

If you loaded the example data we might need to bump that data set with the new columns as well (as it tends to lag behind a bit). I can probably do that in the next day or so.

**Comment by sfc-gh-pkommini** [comment]
it's not an existing database. I loaded the sample data and was able to fix the issue by doing a `dispatch database upgrade` on the dispatch_web_1 container.

We're evaluating it for us on our team at Snowflake and were wondering if there's a way to run the latest migrations before starting the web app during the docker orchestration itself.

**Comment by kevgliss** [comment]
Yeah, in general, I'm not a huge fan of running migrations automatically (can result in data loss/corruption).

But in this scenario I think we can do two things:

- Remove the following conditional (always run the migrations)  https://github.com/Netflix/dispatch-docker/blob/master/install.sh#L133

- Add a script such that we can easily run migrations on the existing .dump file whenever we do a release. Which should help prevent it from lagging behind.

**Comment by kevgliss** [comment]
See: #88 number 1

**Comment by kevgliss** [comment]
See: https://github.com/Netflix/dispatch/pull/676/files for number 2.

Let us know if you run into any other issues.

**Comment by neo4u** [comment]
Will test and close out this issue in a bit.

**Comment by neo4u** [comment]
Thanks a ton for the response!

**Comment by sfc-gh-pkommini** [comment]
Wrong profile sorry. Thanks again!

**Comment by sfc-gh-pkommini** [comment]
I ran the install.sh again and I get the following error:

```
Example data loaded. Navigate to /register and create a new user.
Running standard database migrations...
Creating dispatch_core_1 ... done
Creating dispatch_web_run ... done
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/alembic/script/base.py", line 162, in _catch_revision_errors
    yield
  File "/usr/local/lib/python3.8/site-packages/alembic/script/base.py", line 364, in _upgrade_revs
    revs = list(revs)
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 765, in _iterate_revisions
    requested_lowers = self.get_revisions(lower)
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 319, in get_revisions
    return sum([self.get_revisions(id_elem) for id_elem in id_], ())
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 319, in <listcomp>
    return sum([self.get_revisions(id_elem) for id_elem in id_], ())
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 322, in get_revisions
    return tuple(
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 323, in <genexpr>
    self._revision_for_ident(rev_id, branch_label)
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 386, in _revision_for_ident
    raise ResolutionError(
alembic.script.revision.ResolutionError: No such revision or branch '1c3f808aa00c'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/dispatch", line 8, in <module>
    sys.exit(entrypoint())
  File "/usr/local/lib/python3.8/site-packages/dispatch/cli.py", line 569, in entrypoint
    dispatch_cli()
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/dispatch/cli.py", line 314, in upgrade_database
    alembic_command.upgrade(alembic_cfg, revision, sql=sql, tag=tag)
  File "/usr/local/lib/python3.8/site-packages/alembic/command.py", line 298, in upgrade
    script.run_env()
  File "/usr/local/lib/python3.8/site-packages/alembic/script/base.py", line 489, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/usr/local/lib/python3.8/site-packages/alembic/util/pyfiles.py", line 98, in load_python_file
    module = load_module_py(module_id, path)
  File "/usr/local/lib/python3.8/site-packages/alembic/util/compat.py", line 184, in load_module_py
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/lib/python3.8/site-packages/dispatch/alembic/env.py", line 83, in <module>
    run_migrations_online()
  File "/usr/local/lib/python3.8/site-packages/dispatch/alembic/env.py", line 77, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/usr/local/lib/python3.8/site-packages/alembic/runtime/environment.py", line 846, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/usr/local/lib/python3.8/site-packages/alembic/runtime/migration.py", line 511, in run_migrations
    for step in self._migrations_fn(heads, self):
  File "/usr/local/lib/python3.8/site-packages/alembic/command.py", line 287, in upgrade
    return script._upgrade_revs(revision, rev)
  File "/usr/local/lib/python3.8/site-packages/alembic/script/base.py", line 365, in _upgrade_revs
    return [
  File "/usr/local/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.8/site-packages/alembic/script/base.py", line 194, in _catch_revision_errors
    compat.raise_from_cause(util.CommandError(resolution))
  File "/usr/local/lib/python3.8/site-packages/alembic/util/compat.py", line 308, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=exc_value)
  File "/usr/local/lib/python3.8/site-packages/alembic/util/compat.py", line 301, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.8/site-packages/alembic/script/base.py", line 162, in _catch_revision_errors
    yield
  File "/usr/local/lib/python3.8/site-packages/alembic/script/base.py", line 364, in _upgrade_revs
    revs = list(revs)
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 765, in _iterate_revisions
    requested_lowers = self.get_revisions(lower)
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 319, in get_revisions
    return sum([self.get_revisions(id_elem) for id_elem in id_], ())
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 319, in <listcomp>
    return sum([self.get_revisions(id_elem) for id_elem in id_], ())
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 322, in get_revisions
    return tuple(
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 323, in <genexpr>
    self._revision_for_ident(rev_id, branch_label)
  File "/usr/local/lib/python3.8/site-packages/alembic/script/revision.py", line 386, in _revision_for_ident
    raise ResolutionError(
alembic.util.exc.CommandError: Can't locate revision identified by '1c3f808aa00c'
Cleaning up...
```

**Comment by sfc-gh-pkommini** [comment]
Do you think we can use `upgrade head` to let alembic know just to upgrade to the latest revision as recommended [here](https://alembic.sqlalchemy.org/en/latest/tutorial.html#running-our-first-migration)?

**Comment by sfc-gh-pkommini** [comment]
Is there any way to specify `head` to the dispach CLI?

**Comment by sfc-gh-pkommini** [comment]
Tried doing an `database init` before the upgrade too. That didn't work either.

**Comment by kevgliss** [comment]
Ah, we are about to create a new release (with new migrations) you have a couple of options:

1) Wait for tomorrow when I cut the release
2) Update your docker-compose to point to master here: https://github.com/Netflix/dispatch-docker/blob/master/docker-compose.yml#L22 (just remove the tag information)
3) Fix the install curl to point to the last releases branch instead of master here: https://github.com/Netflix/dispatch-docker/blob/master/install.sh#L122

We should do number 3 regardless but I don't know the URL magic off the top of my head.

**Comment by sfc-gh-pkommini** [comment]
actually I found a possible fix. I can send a diff later today. Let me know if the change is okay with you. I tested it locally and it seems to work.

line 118 - 136 of install.sh to:

```else
  read -p "Do you want to load example data (WARNING: this will remove all existing database data) (y/N)?" CONT
  if [ "$CONT" = "y" ]; then
    echo "Downloading example data from Dispatch repository..."
    curl -O https://raw.githubusercontent.com/Netflix/dispatch/master/data/dispatch-sample-data.dump
    export PGPASSWORD='dispatch'
    echo "Dropping database dispatch if it already exists..."
    dropdb -h localhost -p 5432 -U dispatch dispatch --if-exists
    echo "Creating dispatch database..."
    createdb -h localhost -p 5432 -U dispatch dispatch
    echo "Init schema."
    docker-compose run --rm web database init
    echo "Loading example data to the database..."
    psql -h localhost -p 5432 -U dispatch -d dispatch -f ./dispatch-sample-data.dump
    echo "Example data loaded. Navigate to /register and create a new user."
  fi
  echo "Running standard database migrations..."
  docker-compose run --rm web database upgrade
fi```

basically just added the init to create the alembic_version table.

**Comment by sfc-gh-pkommini** [comment]
Sure yeah. I'll wait till tomorrow. Thank you. I tried some things that didn't work.

**Comment by kevgliss** [comment]
I merged #90 this should help with any mismatches, we will update the latest tag to the most current version.

**Comment by sfc-gh-pkommini** [comment]
It works perfectly now! Thanks again. :)
